packages/api-client/src/client/index.ts

```diff
@@ -70,6 +74,11 @@ function buildConfig(
     if (fetch) newConfig.fetch = fetch;
     if (interceptRequest) newConfig.interceptRequest = interceptRequest;
     if (interceptResponse) newConfig.interceptResponse = interceptResponse;
+    if (hideDeprecatedWarning) {
+        newConfig.hideDeprecatedWarning = hideDeprecatedWarning;
+    } else {
+        newConfig.hideDeprecatedWarning = false;
+    }
```

Comment:
Since it's a boolean, we could keep the same runtime safety with this I think
```suggestion
        newConfig.hideDeprecatedWarning = Boolean(hideDeprecatedWarning);
```